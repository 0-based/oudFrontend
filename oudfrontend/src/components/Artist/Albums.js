import React, { Component } from "react";
import { Link } from "react-router-dom";
import MusicCard from "./../MusicCard/MusicCard";
import { getRequest } from "./../../utils/requester";
import { base } from "./../../config/environment";
class Albums extends Component {
  constructor(props) {
    super(props);
    this.state = {
      albums: [],
      items: [],
      limit: 0,
      seeMore: false,
    };
  }
  componentDidMount() {
    getRequest(`${base}/artists/${this.props.artistId}/albums`)
      .then((response) => {
        this.setState({
          albums: response.data.items,
          limit: response.data.limit,
        });
        this.constructAlbumItems(this.state.albums);
        console.log("fetched artist albums");
        console.log(response);
      })
      .catch((error) => {
        console.log(error.response);
      });
  }
  constructAlbumItems = (albums) => {
    let items = [];
    albums.forEach((album) => {
      const item = {
        id: album._id,
        name: album.name,
        owner: album.artists[0].displayName,
        collaborative: album.artists[0].displayName,
        description: "Album for the artist",
        isPublic: true,
        type: "album",
        image: album.image,
      };
      items.push(item);
    });
    this.setState({
      items: items,
    });
    console.log("album items: ");
    console.log(this.state.items);
  };
  handleSeeMore = () => {};
  pause = () => {};
  resume = () => {};
  addToQueue = () => {};
  addToPlaylist = () => {
    this.setState({ displayAdd: true });
  };
  closeAddToPlaylist = () => {
    this.setState({ displayAdd: false });
  };
  render() {
    return (
      <div className="artis-overview">
        <div className="overview-title">
          <h5 data-test="title" style={{ width: "50%" }}>
            Popular
          </h5>
          <div className="SEE-ALL-Overview" onClick={this.handleSeeMore}>
            See All
          </div>
        </div>
        <div className="wrapper" data-testid="first-wrapper">
          <div className="wrapper_section_2" data-testid="second-wrapper">
            <div className="cards" data-testid="cards-wrapper">
              {this.state.items.map((item, index) => {
                console.log("item:");
                console.log(item);
                return <MusicCard item={item} key={item.id} playBtn={true} />;
              })}
            </div>
          </div>
        </div>
      </div>
    );
  }
}
export default Albums;
