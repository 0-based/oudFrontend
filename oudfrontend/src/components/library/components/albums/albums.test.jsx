import React from 'React'
import Albums from './albums.jsx';
import Enzyme, {shallow} from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'

Enzyme.configure({adapter: new EnzymeAdapter()});


const setup = () =>{
    return shallow(<Albums/>);
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-testid="${val}"]`);
}
/*
Rendering tests
*/

describe('albums Component', ()=>{
    describe('testing library albums renders Correctly', ()=>{
        let component;
        beforeEach (()=>{
            component = setup();
        })
        it('renders albums component', ()=>{
            const wrapper = findByTestAttr(component, "albums");
            expect(wrapper.length).toBe(1);
        });
        it('renders title component', ()=>{
            const wrapper = findByTestAttr(component, "title");
            expect(wrapper.length).toBe(1);
            expect(wrapper.text()).toBe('Saved ALBUMS')
        });
        it('renders 1 card component', ()=>{
            component.setState({recieved:true , albums:[1]})
            const wrapper = findByTestAttr(component, "cards");
            expect(wrapper.length).toBe(1);
        });
        it('renders 2 cards component', ()=>{
            component.setState({recieved:true , albums:[1, 1]})
            const wrapper = findByTestAttr(component, "cards");
            expect(wrapper.length).toBe(2);
        });
        it('renders no cards component when recieved is false', ()=>{
            component.setState({albums:[1, 1]})
            const wrapper = findByTestAttr(component, "cards");
            expect(wrapper.length).toBe(0);
        });
        it('does not render loading component when recieved is true', ()=>{
            component.setState({recieved:true});
            const wrapper = findByTestAttr(component, "loading"); 
            expect(wrapper.length).toBe(0);
        });
        it('renders loading component when recieved is false', ()=>{
            component.setState({recieved:false});
            const wrapper = findByTestAttr(component, "loading"); 
            expect(wrapper.length).toBe(1);
        });
        
    });
    
});