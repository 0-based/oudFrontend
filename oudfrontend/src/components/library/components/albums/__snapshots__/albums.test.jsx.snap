// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`albums Component snapshot test renders correctly 1`] = `
<div
  data-testid="albums"
>
  <h1
    data-testid="title"
  >
    Saved ALBUMS
  </h1>
  <div
    className="spinner"
  >
    <div
      className="rect1"
    />
    <div
      className="rect2"
    />
    <div
      className="rect3"
    />
    <div
      className="rect4"
    />
    <div
      className="rect5"
    />
  </div>
</div>
`;
